Add `Separator` docs (#1398)
hashicorp/design-system: website/docs/components/separator/partials/guidelines/guidelines.md

## Placement

Use separators to distinguish between sections on a page visually. Use them sparingly and purposefully to avoid cluttering the UI and distracting users from the main content.

!!! Do

Use to create a clear break between paragraphs, blocks, or sections.

![Example of the separator component used correctly between content sections] (/assets/components/separator/separator-do.png)

!!!

!!! Dont

Don’t use to separate headings and body text. Instead, emphasize hierarchy by using the proper [Display font style](https://helios.hashicorp.design/foundations/typography?tab=code#font-styles). 

![Example of the separator component used improperly, causing clutter and disrupting content flow."](/assets/components/separator/separator-dont.png)

!!!

## Spacing

We recommend using meaningful spacing between content elements, allowing users to scan and understand information quickly.

> - Use the '0px' spacing option for a tightly integrated layout without any visible gaps between elements.
> - Use the '24px' spacing option to visually distinguish between different sections, making it simpler for users to navigate and understand the content.

Reviewer: ```suggestion
- Use the `0px` spacing option for a tightly integrated layout without any visible gaps between elements.
- Use the `24px` spacing option to visually distinguish between different sections, making it simpler for users to navigate and understand the content.
```

## Placement

Use separators to distinguish between sections on a page visually. Use them sparingly and purposefully to avoid cluttering the UI and distracting users from the main content.

!!! Do

Use to create a clear break between paragraphs, blocks, or sections.

> ![Example of a leading icon in the tag](/assets/components/separator/separator-do.png)

Reviewer 2: The _alt_ text seems to be not correct (probably copy&pasted the code from another component, better to check all the image instances in this PR)

!!!

!!! Dont

> ![Example of a leading icon in the tag](/assets/components/separator/separator-dont.png)

Reviewer 2: As "consumer" I am not sure how to interpret this "don't" image: to me it seems very similar to the "do" one, I am not sure what I should **not** do.

Reviewer: @didoo This might've been before the update to the 'Do & Don't' section, but what do you think now with the added description? [Separator Website Link](https://hds-website-git-alex-ju-add-separator-docs-hashicorp.vercel.app/components/separator)

Reviewer: @heatherlarsen Yes, I have some from the light audit I can fix for the examples. Let me get those in.

!!!

## Spacing

We recommend using meaningful spacing between content elements, allowing users to scan and understand information quickly.

> ### 0px spacing

Reviewer 2: ```suggestion
### `0px` spacing
```

   - Use the `0px` spacing option for a tightly integrated layout without any visible gaps between elements.

>    - Avoid using the `0px` spacing option when the content sections need clear visual separation or when visual clutter is possible.

Reviewer 2: Personally, for better readability, I would remove the "Avoid" paragraph here and in the section below (since they're saying the same thing of the "suggested" ones, only for the other variant).
Essentially I would merge the
- "use 0px" under the `0px` with "avoid 0px" under the `24px`
- "use 24px" under the `24px` with "avoid 24px" under the `0px`

Reviewer 3: I see what @didoo is saying here. It's almost as if there's no need to have the two separate sections. Could you just have one bulleted list under the spacing heading with all the details? Something like: 

> We recommend using meaningful spacing between content elements, allowing users to scan and understand information quickly.
> 
> - Use the `0px` spacing option for a tightly integrated layout without any visible gaps between elements.
> - Use the `24px` spacing option to visually distinguish between different sections, making it simpler for users to navigate and understand the content.

Reviewer: I do like this approach, Its cleaner. I'll update it now! Thank you @didoo and @heatherlarsen

> ### 24px spacing

Reviewer 2: ```suggestion
### `24px` spacing
```

## Placement

Use separators to distinguish between sections on a page visually. Use them sparingly and purposefully to avoid cluttering the UI and distracting users from the main content.

!!! Do

Use to create a clear break between paragraphs, blocks, or sections.

> ![Example of a leading icon in the tag](/assets/components/separator/separator-do.png)

Reviewer: ```suggestion
![Example of separator component used correctly between content sections (/assets/components/separator/separator-do.png)
```

> ![Example of the separator component used correctly between content sections (/assets/components/separator/separator-do.png)

Reviewer: ```suggestion
![Example of the separator component used correctly between content sections] (/assets/components/separator/separator-do.png)
```

!!!

!!! Dont

Don’t use to separate headings and body text. Instead, emphasize hierarchy by using the proper [Display font style](https://helios.hashicorp.design/foundations/typography?tab=code#font-styles). 

> ![Example of a leading icon in the tag](/assets/components/separator/separator-dont.png)

Reviewer: ```suggestion
![Example of the separator component used improperly, causing clutter and disrupting content flow."](/assets/components/separator/separator-dont.png)
```

!!!

## Spacing

We recommend using meaningful spacing between content elements, allowing users to scan and understand information quickly.

> ### `0px` spacing
>    - Use the `0px` spacing option for a tightly integrated layout without any visible gaps between elements.
>    - Avoid using the `0px` spacing option when the content sections need clear visual separation or when visual clutter is possible.
> 
> ### `24px` spacing
>    - This option assists in visually distinguishing between different sections, making it simpler for users to navigate and understand the content.
>    - Similarly, avoid using the `24px` spacing option when you require a tightly integrated layout or when limited vertical space is available.

Reviewer: ```suggestion
### `0px` spacing
   - Use the `0px` spacing option for a tightly integrated layout without any visible gaps between elements.
  - Avoid using the `24px` spacing option when you require a tightly integrated layout or when limited vertical space is available.

### `24px` spacing
   - This option assists in visually distinguishing between different sections, making it simpler for users to navigate and understand the content.
- Similarly, avoid using the `0px` spacing option when the content sections need clear visual separation or when visual clutter is possible.
```

> We recommend using meaningful spacing between content elements, allowing users to scan and understand information quickly.
> 
> ### `0px` spacing
>    - Use the `0px` spacing option for a tightly integrated layout without any visible gaps between elements.
>   - Avoid using the `24px` spacing option when you require a tightly integrated layout or when limited vertical space is available.
> 
> ### `24px` spacing
>    - This option assists in visually distinguishing between different sections, making it simpler for users to navigate and understand the content.
> - Similarly, avoid using the `0px` spacing option when the content sections need clear visual separation or when visual clutter is possible.

Reviewer: ```suggestion
We recommend using meaningful spacing between content elements, allowing users to scan and understand information quickly.

- Use the '0px' spacing option for a tightly integrated layout without any visible gaps between elements.
- Use the '24px' spacing option to visually distinguish between different sections, making it simpler for users to navigate and understand the content.
```

> ## Usage
> 
> ### When to use
> 
> - Use to visually separate different parts or sections on a page.
> - Use it to create a clear break between paragraphs, blocks, or sections.
> 
> ### When not to use
> 
> - Don’t use to separate headings and body text. Instead, use the proper [Display token](https://github.com/hashicorp/design-system/issues/262) to emphasize hierarchy. 

Reviewer 3: This may be an outlier component that doesn't benefit from the usage information in this structure as this content feels more appropriate as a part of the "Placement" section. I've provided a few suggestions below about where it could go instead.
```suggestion

```

## Placement

> We recommend using meaningful spacing between content elements, allowing users to scan and understand information quickly. Use separators sparingly and purposefully to avoid cluttering the UI and distracting users from the main content.

Reviewer 3: Suggesting a move of the spacing sentence to the proper section and adding some content from the "Usage" section. 

```suggestion
Use separators to distinguish between sections on a page visually. Use them sparingly and purposefully to avoid cluttering the UI and distracting users from the main content.
```

> !!! Do
> 
> ![Example of a leading icon in the tag](/assets/components/separator/separator-do.png)
> 
> !!!

Reviewer 3: We've been trying to add content within the do/don't blocks to help support the image. This is a good spot for your statements from the "Usage" section above. 

```suggestion
!!! Do

Use to create a clear break between paragraphs, blocks, or sections.

![Example of a leading icon in the tag](/assets/components/separator/separator-do.png)

!!!
```

> !!! Dont
> 
> ![Example of a leading icon in the tag](/assets/components/separator/separator-dont.png)
> 
> !!!

Reviewer 3: ```suggestion
!!! Dont

Don’t use to separate headings and body text. Instead, use the proper [Display token](https://github.com/hashicorp/design-system/issues/262) to emphasize hierarchy. 

![Example of a leading icon in the tag](/assets/components/separator/separator-dont.png)

!!!
```

Reviewer 2: a couple of points: 
- what does `Display token` mean?
- are you sure the link is correct? it points to a Github issue about  Flight icons

Reviewer 3: @didoo Fair questions, I updated the suggestion. wdyt?

Reviewer 2: 👍

> ### Spacing

Reviewer 3: ```suggestion
## Spacing
```

> 

Reviewer 3: ```suggestion

We recommend using meaningful spacing between content elements, allowing users to scan and understand information quickly.

```

> ##### 0px Spacing:

Reviewer 3: Headings need to go in sequential order to comply with accessibility best practices. We also use sentence case and don't include punctuation in our headings.

```suggestion
### 0px spacing
```

>    - This option works well when you want adjacent components to flow together smoothly without any interruptions in the visual presentation.

Reviewer 3: By nature, this component will always create an interruption in the visual presentation.

```suggestion

```

>    - Use the 0px spacing option for a tightly integrated layout without any visible gaps between elements.

Reviewer 2: ```suggestion
   - Use the `0px` spacing option for a tightly integrated layout without any visible gaps between elements.
```

>    - Avoid using the 0px spacing option when the content sections need clear visual separation or when visual clutter is possible.

Reviewer 2: ```suggestion
   - Avoid using the `0px` spacing option when the content sections need clear visual separation or when visual clutter is possible.
```

> ##### 24px Spacing:

Reviewer 3: ```suggestion
### 24px spacing
```

   - This option assists in visually distinguishing between different sections, making it simpler for users to navigate and understand the content.

>    - Use the 24px spacing option to create a clear visual distinction and breathing space between content sections.

Reviewer 3: Feels like a duplicate of the content above.

```suggestion

```

>    - Use the 24px spacing option to create a clear visual distinction and breathing space between content sections.

Reviewer 2: ```suggestion
   - Use the `24px` spacing option to create a clear visual distinction and breathing space between content sections.
```

>    - Similarly, avoid using the 24px spacing option when you require a tightly integrated layout or when limited vertical space is available.

Reviewer 3: I'm curious about what you mean by "when limited vertical space is available"...can you clarify? It's a common myth that users don't scroll, so I'm not sure we need to really worry about this. 

```suggestion
   - Avoid using the 24px spacing option when you require a tightly integrated layout or when limited vertical space is available.
```

Reviewer: Yeah, looking at again, I see what you mean. I considered when consumers must occupy all the real estate on the screen. So adding that was supposed to communicate compact interfaces / aim for maximizing content within a confined layout.

>    - Similarly, avoid using the 24px spacing option when you require a tightly integrated layout or when limited vertical space is available.

Reviewer 2: ```suggestion
   - Similarly, avoid using the `24px` spacing option when you require a tightly integrated layout or when limited vertical space is available.
```

> !!! Info
> 
> Use these spacing options judiciously, depending on the specific design requirements and the desired user experience.
> 
> !!!

Reviewer 3: This content isn't providing value to the consumer, as it's implied with all of our guidance.

```suggestion

```
